packages/core/echo/echo-db/src/core-db/object-core.ts

```diff
@@ -367,6 +368,21 @@ export class ObjectCore {
   setDeleted(value: boolean) {
     this._setRaw([SYSTEM_NAMESPACE, 'deleted'], value);
   }
+
+  toPlainObject(): CommonObjectData & Record<string, any> {
+    let data = this.getDecoded([DATA_NAMESPACE]);
+    if (typeof data !== 'object') {
+      log.error('Corrupted object data property', { type: typeof data });
+      data = {};
```

Comment:
maybe in such cases move the stored data to some "quarantine" namespace, so that it might be restored / migrated / examined later